tests/model_registry/model_catalog/utils.py

```diff
@@ -1009,6 +1009,120 @@ def validate_items_sorted_correctly(items: list[dict], field: str, order: str) -
         raise ValueError(f"Invalid sort order: {order}")
 
 
+def verify_custom_properties_sorted(items: list[dict], property_field: str, sort_order: str) -> bool:
+    """
+    Verify if a list of items is sorted by a specific custom property value.
+
+    Expected sorting behavior:
+    1. Items WITH the custom property appear first, sorted by property value (ASC or DESC)
+    2. Items WITHOUT the custom property appear after, sorted by ID in ASC order
+
+    Args:
+        items: List of artifact items (with 'id' and 'customProperties')
+        property_field: Property field to sort by (e.g., "e2e_p90.double_value")
+        sort_order: "ASC" or "DESC" (applies only to items with the property)
+
+    Returns:
+        True if sorted correctly, False otherwise
+    """
+    if not items:
+        return True
```

Comment:
We are saying if there is no artifacts, we did not sort, but we are saying sorting works, without testing. This would hide product bugs, if we are unlucky in picking a model with multiple artifacts.